source/flow.rst

```diff
@@ -10,4 +10,438 @@ and events generated by devices.
 .. note::
    - Who is this for: entry-level users
    - Level: basic
-   - Reading time: X m
+   - Reading time: 10 min
+
+Dojot nodes
+-----------
+
+.. contents::
+  :local:
+
+Device in
+****
+
+.. _device_in_node:
+.. image:: images/nodes/device_node.png
+    :width: 20%
+    :align: left
+    :alt: device_node
+
+| This node determine an especific device to be the entry-point of a flow.
+| To configure the device in node, a window like :numref:`device_in_cfg` will be
+displayed.
+
+.. _device_in_cfg:
+.. figure:: images/nodes/device_node_cfg.png
+    :width: 50%
+    :align: center
+    :alt: devicein_node_cfg
+
+    : Device in configuration window
+
+Fields:
+
+* **Name** *(optional)*: Name of the node
+* **Device** *(required)*: The *dojot* device that will trigger the flow
+* **Status** *(required)*: *exclude device status changes* will not use device status changes (online, offline) to trigger the flow. On the other hand, *include devices status changes* will use these status to trigger the flow.
+
+.. note::
+    If the the device that triggers a flow is removed, the flow becomes invalid.
+
+Device template in
+***********
+
+.. _devicetemplate_in_node:
+.. image:: images/nodes/devicetemplate_node.png
+    :width: 20%
+    :align: left
+    :alt: devicetemplatein_node
+
+This node will make that a flow get triggered by devices that are composed by a certain
+template. If the device template that is configured in **device template in** node is template A, all devices 
+that are composed with template A will trigger the flow. For example: *device1* is composed by templates [A,B], 
+*device2* by template A and *device3* by template B. Then, in that scenario, only messages from *device1* and
+*device2* will initiate the flow, because template A is one of the templates that compose those devices.
+
+.. _devicetemplate_in_node:
+.. figure:: images/nodes/devicetemplate_node_cfg.png
+    :width: 50%
+    :align: center
+    :alt: devicetemplatein_node
+
+    : Device template in configuration window
+
+Fields:
+
+* **Name** *(optional)*: Name of the node.
+* **Device** *(required)*: The *dojot* device that will trigger the flow.
+* **Status** *(required)*: Choose if devices status changes will trigger or not the flow.
+
+http
+****
+
+.. _http_node:
+.. image:: images/nodes/http_node.png
+    :width: 20%
+    :align: left
+    :alt: http_node
+
+This node sends an http request to a given address, and, then, it can forward the response to the next node in the flow.
+
+.. _http_in_node:
+.. figure:: images/nodes/http_node_cfg.png
+    :width: 50%
+    :align: center
+    :alt: httpin_node
+
+    : Device template in configuration window
+
+Fields:
+
+* **Method** *(required)*: The http method (GET, POST, etc...).
+* **URL** *(required)*: The URL that will receive the http request
+* **Request body** *(required)*: Variable that contains the request body. This value can be assigned to the variable using the **template node**.
```

Comment:
Maybe a ',for example.' in the end, because it's a suggestion.

Reply:
ok